python/ray/tests/test_runtime_env.py

```diff
@@ -286,6 +287,76 @@ def f():
         ray.get(a.ready.remote())
 
 
+def test_runtime_env_rescheduling(shutdown_only):
+    @ray.remote
+    class A:
+        def get_node_name(self):
+            return os.environ["NODE_NAME"]
+
+        def kill(self):
+            os._exit(0)
+
+    @ray.remote
+    def get_node_name():
+        return os.environ["NODE_NAME"]
+
+    runtime_env = {"env_vars": {"TF_WARNINGS": "none"}}
+    # Runtime env setup will fail at node1, and retry at node2
+    os.environ["_RAY_RUNTIME_ENV_FAIL_AT_NODE_NAME"] = "1"
+
+    cluster = Cluster()
+    cluster.add_node(env_vars={"NODE_NAME": "1"})
+    cluster.add_node(env_vars={"NODE_NAME": "2"})
+    cluster.connect()
+
+    # test task
+    for i in range(10):
+        node_name = ray.get(
+            get_node_name.options(runtime_env=runtime_env).remote())
+        assert node_name == "2"
+    # test actor
```

Comment:
Btw, how does it work without modifying actor manager code? I thought we should modify task spec for actor creation task within gcs?